mpcs/src/basefold.rs

```diff
@@ -320,154 +312,379 @@ where
     }
 
     /// Open a batch of polynomial commitments at several points.
-    /// The current version only supports one polynomial per commitment.
-    /// Because otherwise it is complex to match the polynomials and
-    /// the commitments, and because currently this high flexibility is
-    /// not very useful in ceno.
     fn batch_open(
-        _pp: &Self::ProverParam,
-        _polys: &[ArcMultilinearExtension<E>],
-        _comms: &[Self::CommitmentWithWitness],
-        _points: &[Vec<E>],
-        _evals: &[Evaluation<E>],
-        _transcript: &mut impl Transcript<E>,
-    ) -> Result<Self::Proof, Error> {
-        unimplemented!()
-    }
-
-    /// This is a simple version of batch open:
-    /// 1. Open at one point
-    /// 2. All the polynomials share the same commitment and have the same
-    ///    number of variables.
-    /// 3. The point is already a random point generated by a sum-check.
-    fn simple_batch_open(
         pp: &Self::ProverParam,
-        polys: &[ArcMultilinearExtension<E>],
-        comm: &Self::CommitmentWithWitness,
-        point: &[E],
-        evals: &[E],
+        num_instances: &[(usize, usize)],
+        fixed_comms: Option<&Self::CommitmentWithWitness>,
+        witin_comms: &Self::CommitmentWithWitness,
+        points: &[Point<E>],
+        // TODO this is only for debug purpose
+        evals: &[Vec<E>],
+        circuit_num_polys: &[(usize, usize)],
```

Comment:
why not change the type of `circuit_num_polys` to `BTreeMap<usize, (usize, usize)>` as the matrices/circuit's witness inside `witin_comms` are indexed by the circuit index?

Reply:
because `circuit_num_polys` deterministic and static information came from verifier key. Thus I feel a static information should avoid using `BTreeMap` which make room to include some invalid key. Actually I feel in the end we should change this to fixed size array, and so as other part.